src/SDKs/Azure.Core/data-plane/Azure.Core.Tests/AvoidSyncInterceptor.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Castle.DynamicProxy;
+
+namespace Azure.Core.Tests
+{
+    /// <summary>
+    /// Ensures that tests sticks to calling only async methods
+    /// </summary>
+    public class AvoidSyncInterceptor : IInterceptor
+    {
+        public void Intercept(IInvocation invocation)
+        {
+            if (!invocation.Method.Name.EndsWith("Async"))
+            {
+                throw new InvalidOperationException("Async method call expected");
```

Comment:
Wait, won't this fail for things like GetFooClient()?  Should we also make sure there's a similarly named method with an Async suffix before we throw?

Reply:
Fixed